llvm/lib/MC/MCFragment.cpp

```diff
@@ -203,59 +203,57 @@ MCFragment::MCFragment(FragmentType Kind, bool HasInstructions)
 
 void MCFragment::destroy() {
   // First check if we are the sentinel.
-  if (Kind == FragmentType(~0)) {
```

Comment:
`~MCFragment();`?

Reply:
It looks like the `Kind == FragmentType(~0)` code path is now dead. I'll just delete this condition.